lib/jxl/base/status.h

```diff
@@ -321,6 +324,84 @@ inline JXL_FORMAT(2, 3) Status
   return status;
 }
 
+template <typename T>
+class JXL_MUST_USE_RESULT StatusOr {
+  union Storage {
+    char dummy_;
+    T data_;
+  } storage_;
+
+  static_assert(!std::is_convertible<StatusCode, T>::value &&
+                    !std::is_convertible<T, StatusCode>::value,
+                "You cannot make a StatusOr with a type convertible from or to "
```

Comment:
```suggestion
                "You can only make a StatusOr with a type convertible from or to "
```

Reply:
I think your suggestion changes the meaning of what the error says :-)

Reply:
Oh, I misunderstood, I thought the template param was to eventually use it for both jxl::Status and C API error codes, but the template param is for the type that may fail to construct due to OOM. Nevermind!